src/ascii_char.rs

```diff
@@ -297,165 +288,323 @@ impl AsciiChar {
+            (b'0'..=b'9', 0..=36) => u32::from(self as u8 - b'0') < radix,
+            (b'a'..=b'z', 11..=36) => u32::from(self as u8 - b'a') < radix - 10,
+            (b'A'..=b'Z', 11..=36) => u32::from(self as u8 - b'A') < radix - 10,
+            (_, 0..=36) => false,
+            (_, _) => panic!("radixes greater than 36 are not supported"),
+        }
     }
 
     /// Check if the character is a number (0-9)
+    ///
+    /// # Examples
+    /// ```
+    /// # use ascii::AsciiChar;
+    /// assert_eq!(AsciiChar::new('0').is_ascii_digit(), true);
+    /// assert_eq!(AsciiChar::new('9').is_ascii_digit(), true);
+    /// assert_eq!(AsciiChar::new('a').is_ascii_digit(), false);
+    /// assert_eq!(AsciiChar::new('A').is_ascii_digit(), false);
+    /// assert_eq!(AsciiChar::new('/').is_ascii_digit(), false);
+    /// ```
     #[inline]
-    pub fn is_digit(self) -> bool {
-        self >= AsciiChar::_0 && self <= AsciiChar::_9
+    pub const fn is_ascii_digit(&self) -> bool {
+        (*self as u8 >= b'0') & (*self as u8 <= b'9')
     }
 
     /// Check if the character is a letter or number
     #[inline]
-    pub fn is_alphanumeric(self) -> bool {
-        self.is_alphabetic() || self.is_digit()
+    pub const fn is_alphanumeric(self) -> bool {
+        self.is_alphabetic() | self.is_ascii_digit()
+    }
+
+    /// Check if the character is a letter or number
+    ///
+    /// This method is identical to [`is_alphanumeric()`](#method.is_alphanumeric)
+    pub fn is_ascii_alphanumeric(&self) -> bool {
+        self.is_alphanumeric()
     }
 
     /// Check if the character is a space or horizontal tab
+    ///
+    /// # Examples
+    /// ```
+    /// # use ascii::AsciiChar;
+    /// assert!(AsciiChar::Space.is_ascii_blank());
+    /// assert!(AsciiChar::Tab.is_ascii_blank());
+    /// assert!(!AsciiChar::VT.is_ascii_blank());
+    /// assert!(!AsciiChar::LineFeed.is_ascii_blank());
+    /// assert!(!AsciiChar::CarriageReturn.is_ascii_blank());
+    /// assert!(!AsciiChar::FF.is_ascii_blank());
+    /// ```
     #[inline]
-    pub fn is_blank(self) -> bool {
-        self == AsciiChar::Space || self == AsciiChar::Tab
+    pub const fn is_ascii_blank(self) -> bool {
+        (self as u8 == b' ') | (self as u8 == b'\t')
     }
 
-    /// Check if the character is a ' ', '\t', '\n' or '\r'
+    /// Check if the character one of ' ', '\t', '\n', '\r',
+    /// '\0xb' (vertical tab) or '\0xc' (form feed).
     #[inline]
-    pub fn is_whitespace(self) -> bool {
-        self.is_blank() || self == AsciiChar::LineFeed || self == AsciiChar::CarriageReturn
+    pub const fn is_whitespace(self) -> bool {
+        let b = self as u8;
+        self.is_ascii_blank() | (b == b'\n') | (b == b'\r') | (b == 0x0b) | (b == 0x0c)
+    }
+
+    /// Check if the character is a ' ', '\t', '\n', '\r' or '\0xc' (form feed).
+    ///
+    /// This method is NOT identical to `is_whitespace()`.
+    #[inline]
+    pub const fn is_ascii_whitespace(self) -> bool {
```

Comment:
`is_ascii_whitespace` also exists on `&[u8]`, therefore it should take `&self`.

Reply:
OK